.github/workflows/gpuci.yml

```diff
@@ -21,27 +23,30 @@ concurrency:
 jobs:
   # First, we build and push a NeMo-Curator container
   build-container:
-    # "build-container" job is run if the "gpuci" label is added to the PR
-    if: ${{ github.event.label.name == 'gpuci' || github.ref == 'refs/heads/main' }}
+    # "build-container" job is run if the "gpuci" label is added to the PR / merge to main / scheduled run
+    if: ${{ github.event.label.name == 'gpuci' || github.ref == 'refs/heads/main' || github.event_name == 'schedule' }}
     uses: NVIDIA/NeMo-FW-CI-templates/.github/workflows/_build_container.yml@v0.18.0
     with:
       image-name: nemo_curator_container
-      dockerfile: Dockerfile
-      image-label: nemo-curator
+      # Use different Dockerfile based on event type
+      dockerfile: ${{ github.event_name == 'schedule' && 'Dockerfile.nightly' || 'Dockerfile' }}
+      image-label: ${{ github.event_name == 'schedule' && 'nemo-curator-nightly' || 'nemo-curator' }}
       build-args: |
-        IMAGE_LABEL=nemo-curator
+        IMAGE_LABEL=${{ github.event_name == 'schedule' && 'nemo-curator-nightly' || 'nemo-curator' }}
         REPO_URL=https://github.com/${{ github.repository }}.git
         CURATOR_COMMIT=${{ github.sha }}
+        BUILD_TYPE=${{ github.event_name == 'schedule' && 'nightly' || 'stable' }}
       prune-filter-timerange: 24h
+      use_cache: ${{ github.event_name != 'schedule' }}
```

Comment:
Would you mind elaborating on how this section works?

Reply:
Yes, to begin with `${{ }}}` is the [syntax for expressions](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#example).

And then whenever the statement `github.event_name == 'schedule'` is `True` we know we're dealing with nightly (as it runs on schedule) instead of the merge-to-main / PR labeling stuff.

Now going over the individual build args that are passed to Dockerfile
1. IMAGE_LABEL = `nemo-curator-nightly` or `nemo-curator`
2. REPO_URL = `github.com/NVIDIA/NeMo-Curator`
3. CURATOR_COMMIT = `commit_sha`
4. BUILD_TYPE = `nightly` or `stable`

Apart from build args we also don't want to use cache for nightly when building docker images hence we use the conditional.